### PR TITLE
samples: sensor: Add ADXL345 sensor sample

### PR DESCRIPTION
**Pull request by soburi**
Add a basic sample application for ADXL345.
This sample demonstrates how to data fetch and print to the console.

### CODE REVIEW

File: samples/sensor/adxl345/src/main.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+
+void main(void)
+{
+	struct sensor_value accel[3];
+
+	const struct device *dev = device_get_binding(DT_LABEL(DT_INST(0, adi_adxl345)));
```
Comment (gmarull) [inline samples/sensor/adxl345/src/main.c:15]: Please use DEVICE_DT_GET_ONE or ANY.

File: samples/sensor/adxl345/src/main.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	if (dev == NULL) {
+		printk("device_get_binding() failed");
+		return;
+	}
+
+	while (true) {
+		int ret;
+		static const int channels[] = {
```
Comment (gmarull) [inline samples/sensor/adxl345/src/main.c:24]: You can use channel enum here, not int.

File: samples/sensor/adxl345/src/main.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			SENSOR_CHAN_ACCEL_Z,
+		};
+
+		ret = sensor_sample_fetch(dev);
+		if (ret < 0) {
+			printk("sensor_Sample_fetch() failed: %d\n", ret);
+			return;
+		}
+
+		for (int i = 0; i < ARRAY_SIZE(channels); i++) {
```
Comment (gmarull) [inline samples/sensor/adxl345/src/main.c:36]: `size_t i`

File: samples/sensor/adxl345/sample.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+sample:
+  name: ADXL345 Sensor sample
+tests:
+  sample.sensor.adxl345.i2c:
+    harness: console
+    tags: sensors
+    depends_on: i2c
+    integration_platforms:
+      - nucleo_f030r8
+    platform_allow: nucleo_f030r8
```
Comment (gmarull) [inline samples/sensor/adxl345/sample.yaml:10]: Please filter based on sensor dt compatible being enabled. This sample is not board specific.

**Review by gmarull — CHANGES_REQUESTED** [review]
Thanks for the contribution, please check my comments. Doesn’t this depend on #45028 ?

File: samples/sensor/adxl345/src/main.c (hunk 1 above)
Comment (soburi) [reply samples/sensor/adxl345/src/main.c]: Fix it.

File: samples/sensor/adxl345/src/main.c (hunk 2 above)
Comment (soburi) [reply samples/sensor/adxl345/src/main.c]: Fix it.

File: samples/sensor/adxl345/src/main.c (hunk 3 above)
Comment (soburi) [reply samples/sensor/adxl345/src/main.c]: Fix it.

File: samples/sensor/adxl345/sample.yaml (hunk 4 above)
Comment (soburi) [reply samples/sensor/adxl345/sample.yaml]: Updated. Please verify it.

**Comment by soburi** [comment]
Hi, @gmarull 

> Thanks for the contribution, please check my comments. Doesn’t this depend on https://github.com/zephyrproject-rtos/zephyr/pull/45028 ?

Yes. I forgot to notifiy. Sorry.
Update PR description.

**Comment by soburi** [comment]
#45028 is merged. and Rebased this PR.

**Pull request closed without merging**